### PR TITLE
Add CA and leaf cert validations

### PR DESCRIPTION
**Pull request by jondonas**
Adds validations specified here: https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#certificate-requirements

### CODE REVIEW

**Review by shizhMSFT — COMMENTED** [review]
General golang styling comments:
- Document all exported methods
- Coding blocks like
  ```go
  if err := doSomeThing(); err != nil {
    return err
  }
  return nil
  ```
  can be simplified as
  ```go
  return doSomeThing()
  ```
- Acronyms should be all caps. e.g. `CA`, `EKU`, `OID` over `Ca`, `Eku`, `Oid`.
- Add comment and references if the code block is hard to be understood by others.

File: x509/cert_validations.go [hunk 1]
```diff
@@ -0,0 +1,166 @@
...
+func VerifyCodeSigningCertificate(cert *x509.Certificate) error {
+	return verifyCertificate(cert, x509.ExtKeyUsageCodeSigning)
+}
+
+// VerifyTimeStampingCertificate validates time-stamping certificates according to this spec:
+// https://github.com/notaryproject/notaryproject/blob/main/signature-specification.md#leaf-certificates
+func VerifyTimeStampingCertificate(cert *x509.Certificate) error {
+	return verifyCertificate(cert, x509.ExtKeyUsageTimeStamping)
+}
+
```
Comment (priteshbandi) [inline x509/cert_validations.go:31]: 1. Can we please update [`ValidateCertChain`](https://github.com/notaryproject/notation-core-go/blob/main/x509/cert.go#L49) method with this validations?
2. If we do 1st then we dont need to make these methods public.
3. Should we move `ValidateCertChain` from `certs.go` to `cert_validations.go`

File: x509/cert_validations.go [hunk 2]
```diff
@@ -0,0 +1,207 @@
...
+	if cert.KeyUsage&x509.KeyUsageDigitalSignature == 0 {
+		return errors.New("key usage must have the bit positions for digital signature set")
+	}
+	if cert.KeyUsage&x509.KeyUsageCertSign != 0 || cert.KeyUsage&x509.KeyUsageCRLSign != 0 {
+		return errors.New("key usage must not have the bit positions for key cert sign or crl sign set")
+	}
+	return nil
+}
+
+func validateKeyUsagePresent(cert *x509.Certificate) error {
```
[the remark this answers is not in the document]
Comment (jondonas) [reply x509/cert_validations.go:128]: Just for presence yes, but that doesn't tell us KU is marked critical. We need to pull the specific Extension to check that

**Review by priteshbandi — APPROVED** [review]
LGTM

**Comment by jondonas** [comment]
> General golang styling comments:
> 
>     * Document all exported methods
> 
>     * Coding blocks like
>       ```go
>       if err := doSomeThing(); err != nil {
>         return err
>       }
>       return nil
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       can be simplified as
>       ```go
>       return doSomeThing()
>       ```
> 
>     * Acronyms should be all caps. e.g. `CA`, `EKU`, `OID` over `Ca`, `Eku`, `Oid`.
> 
>     * Add comment and references if the code block is hard to be understood by others.



> General golang styling comments:
> 
>     * Document all exported methods
> 
>     * Coding blocks like
>       ```go
>       if err := doSomeThing(); err != nil {
>         return err
>       }
>       return nil
>       ```
>       
>       
>           
>             
>           
>       
>             
>           
>       
>           
>         
>       can be simplified as
>       ```go
>       return doSomeThing()
>       ```
> 
>     * Acronyms should be all caps. e.g. `CA`, `EKU`, `OID` over `Ca`, `Eku`, `Oid`.
> 
>     * Add comment and references if the code block is hard to be understood by others.

Addressed all these comments

**Review by rgnote — APPROVED** [review]

**Review by priteshbandi — APPROVED** [review]
LGTM.

**Pull request merged**